app/main.py

```diff
@@ -2,0 +1,96 @@
+from app.cafe import Cafe
+from app.errors import (NotVaccinatedError, OutdatedVaccineError,
+                        NotWearingMaskError)
+import datetime
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_needed = 0
+    all_vaccinated = True
+
+    for friend in friends:
+        try:
+            if "vaccine" not in friend:
+                all_vaccinated = False
+                continue
+
+            expiration_date = friend["vaccine"]["expiration_date"]
+            if expiration_date < datetime.date.today():
+                all_vaccinated = False
+
+            if not friend.get("wearing_a_mask", False):
+                masks_needed += 1
```

Comment:
Instead of manually incrementing `masks_needed` when a mask is not worn, you should raise a `NotWearingMaskError`. This will allow the `try/except` block to handle the error as intended.